mayastor-test/e2e/nightly/common/util.go

```diff
@@ -0,0 +1,361 @@
+}
+
+func LabelNode(nodename string, label string) {
+	cmd := exec.Command("kubectl", "label", "node", nodename, label)
+	cmd.Dir = ""
+	_, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// Status part of the mayastor volume CRD
+type mayastorVolStatus struct {
+	state  string
+	reason string
+	node   string
+}
+
+func GetMSV(uuid string, dynamicClient * dynamic.Interface) *mayastorVolStatus {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorvolumes",
+	}
+	msv, err := (*dynamicClient).Resource(msvGVR).Namespace("mayastor").Get(context.TODO(), uuid, metav1.GetOptions{})
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	if msv == nil {
+		return nil
+	}
+	status, found, err := unstructured.NestedFieldCopy(msv.Object, "status")
+	if err != nil {
+		fmt.Println(err)
+		return nil
+	}
+
+	if !found {
+		return nil
+	}
+	msVol := mayastorVolStatus{}
+	v := reflect.ValueOf(status)
+	if v.Kind() == reflect.Map {
+		for _, key := range v.MapKeys() {
+			sKey := key.Interface().(string)
+			val := v.MapIndex(key)
+			switch sKey {
+			case "state":
+				msVol.state = val.Interface().(string)
+				break
+			case "reason":
+				msVol.reason = val.Interface().(string)
+				break
+			case "node":
+				msVol.node = val.Interface().(string)
+				break
+			}
+		}
+	}
+	return &msVol
+}
+
+// Check for a deleted Mayastor Volume,
+// the object does not exist if deleted
+func IsMSVDeleted(uuid string, dynamicClient * dynamic.Interface) bool {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorvolumes",
+	}
+
+	msv, err := (*dynamicClient).Resource(msvGVR).Namespace("mayastor").Get(context.TODO(), uuid, metav1.GetOptions{})
+
+	if err != nil {
+		// Unfortunately there is no associated error code so we resort to string comparison
+		if strings.HasPrefix(err.Error(), "mayastorvolumes.openebs.io") &&
+			strings.HasSuffix(err.Error(), " not found") {
+			return true
+		}
+	}
+
+	Expect(err).To(BeNil())
+	Expect(msv).ToNot(BeNil())
+	return false
+}
+
+// Check for a deleted Persistent Volume Claim,
+// either the object does not exist
+// or the status phase is invalid.
+func IsPVCDeleted(volName string, kubeInt * kubernetes.Interface) bool {
+	pvc, err := (*kubeInt).CoreV1().PersistentVolumeClaims("default").Get(context.TODO(), volName, metav1.GetOptions{})
+	if err != nil {
+		// Unfortunately there is no associated error code so we resort to string comparison
+		if strings.HasPrefix(err.Error(), "persistentvolumeclaims") &&
+			strings.HasSuffix(err.Error(), " not found") {
+			return true
+		}
+	}
+	// After the PVC has been deleted it may still accessible, but status phase will be invalid
+	Expect(err).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
+	switch pvc.Status.Phase {
+	case
+		corev1.ClaimBound,
+		corev1.ClaimPending,
+		corev1.ClaimLost:
+		return false
+	default:
+		return true
+	}
```

Comment:
Could we invert this "switch" statement so that we match on the specific "invalid" status phase and return true? Then the default case would catch all other statuses and return false. This way, if an additional status phase is added, we would, by default, return false rather than true.

If possible, this would also be the case for the PV.

Reply:
Ok, will do.